### ISSUE TITLE
Extension icons don't work with Microsoft Store version

### ISSUE THREAD

**Issue by nicoco007** [d2phap/ImageGlass#1847]
### System information:
 - Windows OS version: Windows 11 Pro 23H2 (Build 22631) x64
 - ImageGlass version: 9.0.10.201

### To Reproduce
1. Uninstall regular version of ImageGlass
2. Install ImageGlass from Microsoft Store
3. Go to Settings > File type associations and press "Make default"

### Actual behavior:
Icons in Windows Explorer flicker, indicating something changed, but all registered files still use ImageGlass's logo. If I go to default apps settings, not all extensions are assigned to ImageGlass either, although that seems to be the case with the regular version as well. Pressing "Remove default" then "Make default" again doesn't help.

### Expected behavior:
Image thumbnails use extension-specific icons.

### Screenshots:
![image](https://github.com/d2phap/ImageGlass/assets/1349975/7d3432bf-7ac1-45cc-94fc-40df524e93b6)

### Additional context:
Reinstalling the regular version makes the icons show up as expected.

I think this is related to how Microsoft Store apps store settings in the registry. In the `HKEY_CURRENT_USER\Software\RegisteredApplications` key, I see a value `AppXkycr36dyt7aej1hsxnvpn7vnn8w7m0je` set to `Software\Classes\Local Settings\Software\Microsoft\Windows\CurrentVersion\AppModel\Repository\Packages\9662DuongDieuPhap.ImageGlass_9.0.11.0_x64__xjrmsrdc1fgj6\IMAGEGLASS\Capabilities`, which is very different from what ImageGlass is doing [over here](https://github.com/d2phap/ImageGlass/blob/f5fa3b2fffe3c54596cf35cd4dff97c465ae1c53/Source/Components/ImageGlass.Base/WinApi/ExplorerApi.cs). There's also a value called `9662DuongDieuPhap.ImageGlass_xjrmsrdc1fgj6!IMAGEGLASS` in `PackagedApps` set to the same path.

**Comment by mojoZHU** [comment]
I seem to have also reproduced this issue on the "portable" version that isn't installed with the msi file. No matter what I try, the icons won't appear. I have also tried manually associating them, but this seems to apply the same icon for all file types and not for individual file types

**Comment by ThioJoe** [comment]
**TL;DR: I think the solution is that in the registry location with the file associations that defines the "ProgIDs" for each file extension, there needs to be a unique ProgID (`AppX.....`) for each extension that will then go in HKEY_CLASSES_ROOT so each can define a unique path to a custom icon via the `DefaultIcon` key.  Similar to how with the MSI version there are already entries such as `ImageGlass.AssocFile.TIF` for each filetype there.**

----

Also having the same issue with the Microsoft Store version.

I was digging into how file association icons work with Microsoft Store / Appx packages, and I think the problem is in this registry location:
```
HKEY_CURRENT_USER\Software\Classes\Local Settings\Software\Microsoft\Windows\CurrentVersion\AppModel\Repository\Packages\9662DuongDieuPhap.ImageGlass_9.1.5.0_x64__xjrmsrdc1fgj6\IMAGEGLASS\Capabilities\FileAssociations
```

Each of the file associations is defined with the same "ProgID" I believe it's called, that being  `AppXys8axmqwgf39frtmvts74qj8z4e1w2pf`

<img width="1002" alt="image" src="https://github.com/user-attachments/assets/252d7924-237b-4b78-82bc-2c3dc4b6b225">

If we look up that in HKEY_CLASSES_ROOT here and go to the `DefaultIcon` key:
`HKEY_CLASSES_ROOT\AppXys8axmqwgf39frtmvts74qj8z4e1w2pf\DefaultIcon`

There is a value of `@{9662DuongDieuPhap.ImageGlass_9.1.5.0_x64__xjrmsrdc1fgj6?ms-resource://ImageGlass/Files/Assets/IMAGEGLASS-Square44x44Logo.png}` which leads to that default icon that appears for all filetypes as the icon.

**Then I tested changing this value to just the path of a different PNG file and indeed it changed the image of all file extensions associated with ImageGlass.** 

So that is where it's getting the default icon from, but how do we set custom icons for each file? Well it seems a different "ProgID" (Appx... string) would need to be created for each one, each with a different Default value in the `DefaultIcon` key. At least from what I've seen.

-----

**For example**, take Windows Notepad, the version from the Microsoft Store, which lists the ProgIDs for its file associations at:
```
HKEY_CURRENT_USER\Software\Classes\Local Settings\Software\Microsoft\Windows\CurrentVersion\AppModel\Repository\Packages\Microsoft.WindowsNotepad_11.2407.9.0_x64__8wekyb3d8bbwe\App\Capabilities\FileAssociations
```

<img width="977" alt="image" src="https://github.com/user-attachments/assets/cbba7a08-1dd2-4418-879b-1aa2f24581ac">

Notice how it has multiple `Appx...` ProgID values for different file types, and if you set the default app for an `.ini` file vs a `.txt` file both to Notepad, they have different icons that correspond to the `HKEY_CLASSES_ROOT\` keys for those ProgIDs.

**Comment by pschannes** [comment]
I have the same issue. I have both the Windows store version (called "ImageGlass 9") and the regular version (called "ImageGlass") installed. As soon as I switch the default program for the files to the regular version, the icon extensions work again. Returning to the Windows store version resets the problem to only a single icon showing for all file types. Adding icons to the ext-ico folder as the program suggests doesn't fix the issue.

Edit:
The Windows store version also seems unable to perform the action shown on the screenshot below (make it the default photo viewer program for all files associated) - you have to do it manually for every file extension in the Windows preferences. Whereas the regular version is able to perform this action in one click.

![imgglassbug_01](https://github.com/user-attachments/assets/beb14764-5e90-4fef-a431-cc1b845938a6)
![imgglassbug_02](https://github.com/user-attachments/assets/9d67a458-be66-4f54-8721-b86b18c9774b)

**Comment by ghost** [comment]
The scoop installation method encountered the same problem

**Comment by BigSkyMG** [comment]
Hi, d2phap, is there any update to this issue? I just purchased version 9.2.0.1208, and I am unable to get the extensions icon pack to show in Windows Explorer. It works when I use the version directly from your website, but it doesn't work with the Microsoft Store version. Thank you!

**Comment by Comprehensive-Jason** [comment]
I'm also interested in when this issue is planned to be fixed. I find ImageGlass to be lovely and have paid for the MS Store version, but find this problem to be a significant annoyance.

**Comment by PaulCoddington** [comment]
Also applies to version 10 RC.

**Comment by d2phap** [comment]
Official: 

This is a limitation. The Store release will not support for extension icons. This is because the MSIX package enables virtualization for the registry and file system by default.

I can declare [`unvirtualizedResources`](https://learn.microsoft.com/en-us/windows/uwp/packaging/app-capability-declarations) capability to disable it, but Microsoft will reject on the Store submission. In v10, you can use this feature for the MSIX and ZIP release but it will be disabled for the MSIX Store release.
